### PR TITLE
Update admin.php

### PR DESCRIPTION
**Pull request by smileBeda**
- [Fixed] Wrong option accessed to retrieve the `max-tags` value

> This patch fixes an issue where no matter the value set in the Discourse Publishing Options `max-tags`, it alway allowed only 5 tags maximally. The problem was due to the wrong option name being accessed to retrieve the `max-tags` value.

Props: Richard (Pavillion) for helpful hint on responsible code line.

### CODE REVIEW

**Comment by smileBeda** [comment]
Also, I just realize that now, to denote is the unsafe access of the option value. This should be sanitized, with something like [absint()](https://developer.wordpress.org/reference/functions/absint/), before being used in the code. However as the PR here is not for that purpose I will leave it to maintainers to look at this aspect of the code.

**Comment by angusmcleod** [comment]
@smileBeda Thanks for this.

> Also, I just realize that now, to denote is the unsafe access of the option value. This should be sanitized, with something like [absint()](https://developer.wordpress.org/reference/functions/absint/), before being used in the code.

It's sanitised when it's set.

**Comment by smileBeda** [comment]
Sanitizing when saving to the database _or_ when saving to $variable and escaping when echoing :)
_Even if it is retrieved from the database we control_

This because even if you sanitize when set, another actor could do whatever with it in the database, and when retrieved, we cannot trust it.

_WP would not let it pass_

But, as said this is totally up to you and not the subject of the PR anyway.


Thanks for considering the PR, this it the most important I think from the user perspective.

**Review by angusmcleod — APPROVED** [review]

**Review by Jailyard90Grad — APPROVED** [review]

**Pull request merged**
